Felienne/hedy
# Differentiaties lifespan of recover token from login token

**Description**
We did some digging and I think we figured out the exact functionality of our login tokens. Currently we set a front-end login cookie to expire in **1 year**. While the database saved cookie only has a lifespan of **14 days**. This is no problem as the front-end cookie is verified by the database if it is still valid. When a user logs out we remove the corresponding token in the database, if not the token is automatically removed by Dynamo in 14 days. (This function was turned off, but fixed again together with #2205.) The same lifespan is used for the reset tokens.

In this PR we change this by given the reset tokens a dedicated lifespan, of only 4 hours. We also update this in the mail template to users are aware that this is happening. One last security issue I'm struggling with: If the user is able to retrieve their current login token from the front-end cookie, this one can also be used to reset a password. Do we think this is a problem?


**Fixes**
This PR fixes #2255.

**How to test**
This is a complete back-end test, nothing in functionality should have changed. You can however test it by emptying your `tokens` table in `dev_database.json` and login. Verify that there is one new token available and copy the `ttl` value to a website such as https://www.unixtimestamp.com and verify that the lifespan is indeed 14 days. Then navigate to the `/recover` page and request a new password. (This should work locally with a token return). Verify that there is a new entry in the db with only a ttl of 4 hours.
